### PR TITLE
DPL Analysis: minor cleanups in ASoA

### PR DESCRIPTION
**Pull request by aalkin**

### CODE REVIEW

**Comment by ktf** [comment]
@TimoWilken I think there is something broken with the CI:

```
CMake Error: Generator: execution of make failed. Make command was: /sw/slc8_x86-64/ninja/fortran-v1.8.2.g3b-19/bin/ninja -j 32 install && 
```

is this due to the new container?

**Comment by TimoWilken** [comment]
Hi @ktf, no, the slc8 container was not changed.

**Comment by TimoWilken** [comment]
@ktf looks like the builder actually ran out of memory (the peak around 12:50 is when [the log](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/12201/092025d806d30e0ae6df4d02038ab446b2b8e22e/build_O2_o2-dataflow/fullLog.txt) says it finished on `c03e0332`):

Screenshot_2023-11-03_13-47-59

**Comment by TimoWilken** [comment]
Hi @aalkin, sorry, this keeps OOMing the Linux CI builders, so I'm marking this as draft.

**Comment by aalkin** [comment]
@TimoWilken this is very surprising. I see it also fails with mac CI, which I cannot reproduce locally. I'm not sure how this tiny change can cause anything like this.

**Comment by TimoWilken** [comment]
Hm, e.g. for the first error in the ARM Mac CI:

```
/Volumes/build/alice-ci-workdir/o2/sw/SOURCES/O2/12201/0/Framework/Core/include/Framework/AnalysisDataModel.h:130:1: error: no member named 'FunctionMetadata' in namespace 'o2::framework'
```

`FunctionalMetadata` apparently comes from `Framework/FunctionalHelpers.h`, and you're removing that include, and I can't see another header that includes it transitively in `ASoA.h`.

At least I can see how removing includes might cause problems like this. I don't know how that would cause an OOM on Linux/GCC, though...

File: Framework/Core/src/ASoA.cxx [hunk 1]
```diff
@@ -10,10 +10,9 @@
 // or submit itself to any jurisdiction.
 
 #include "Framework/ASoA.h"
-#include "ArrowDebugHelpers.h"
```
Comment (ktf) [inline Framework/Core/src/ASoA.cxx:13]: Why removing this?

File: Framework/Core/include/Framework/ASoA.h [hunk 2]
```diff
@@ -14,13 +14,10 @@
...
-#include "Framework/CheckTypes.h"
-#include "Framework/FunctionalHelpers.h"
 #include "Framework/CompilerBuiltins.h"
 #include "Framework/Traits.h"
 #include "Framework/Expressions.h"
 #include "Framework/ArrowTypes.h"
-#include "Framework/RuntimeError.h"
 #include "Framework/ArrowTableSlicingCache.h"
 #include "Framework/SliceCache.h"
 #include <arrow/table.h>
```
Comment (ktf) [inline Framework/Core/include/Framework/ASoA.h:23]:
```suggestion
#include <arrow/table.h>
#include <functional>
```

**Review by ktf — CHANGES_REQUESTED** [review]
We should fix at least the missing functional.

**Comment by aalkin** [comment]
@TimoWilken I have not been able to find any reason behind OOM, but the issue with missing `<functional>` include is indeed present when compiling with gcc and older clang. Could you please monitor this PR and turn it off if it causes OOM again? Thanks.

**Comment by TimoWilken** [comment]
Thanks @aalkin, will do! I've also been able to reproduce the OOM locally; I'm running the compilation again now to try and catch which process causes it.

**Comment by aalkin** [comment]
It seems the changes still cause problem for CI, I will close this for now.

**Pull request closed without merging**